File: UserControlForm/UserControlForm/UserControlForm.Web/Modules/ServerTypes/Administration/UserFormEditorForm.ts
import { StringEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface UserFormEditorForm {
    FormName: StringEditor;
    Description: TextAreaEditor;
    Purpose: TextAreaEditor;
    Instructions: TextAreaEditor;
    FormDesign: StringEditor;
}

export class UserFormEditorForm extends PrefixedContext {
    static readonly formKey = 'Administration.UserFormEditor';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!UserFormEditorForm.init)  {
            UserFormEditorForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;

            initFormType(UserFormEditorForm, [
                'FormName', w0,
                'Description', w1,
                'Purpose', w1,
                'Instructions', w1,
                'FormDesign', w0
            ]);
        }
    }
}

File: UserControlForm/UserControlForm/UserControlForm.Web/Modules/Administration/UserFormEditor/UserFormEditorForm.cs
using Serenity.ComponentModel;

namespace UserControlForm.Administration.Forms;

[FormScript("Administration.UserFormEditor")]
[BasedOnRow(typeof(UserFormEditorRow), CheckNames = true)]
public class UserFormEditorForm
{
    [LabelWidth(150)]
    public string FormName { get; set; }
    
    [LabelWidth(150), TextAreaEditor(Rows = 2)]
    public string Description { get; set; }
    
    [LabelWidth(150), TextAreaEditor(Rows = 3)]
    public string Purpose { get; set; }
    
    [LabelWidth(150), TextAreaEditor(Rows = 4)]
    public string Instructions { get; set; }
    
    [HideOnInsert, HideOnUpdate]
    public string FormDesign { get; set; }
}